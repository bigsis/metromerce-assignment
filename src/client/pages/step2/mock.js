export const mockRepo = [
  {
    id: 27,
    name: 'rubinius',
    full_name: 'rubinius/rubinius',
    owner: {
      login: 'rubinius',
      id: 317747,
      avatar_url: 'https://avatars2.githubusercontent.com/u/317747?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/rubinius',
      html_url: 'https://github.com/rubinius',
      followers_url: 'https://api.github.com/users/rubinius/followers',
      following_url:
        'https://api.github.com/users/rubinius/following{/other_user}',
      gists_url: 'https://api.github.com/users/rubinius/gists{/gist_id}',
      starred_url:
        'https://api.github.com/users/rubinius/starred{/owner}{/repo}',
      subscriptions_url: 'https://api.github.com/users/rubinius/subscriptions',
      organizations_url: 'https://api.github.com/users/rubinius/orgs',
      repos_url: 'https://api.github.com/users/rubinius/repos',
      events_url: 'https://api.github.com/users/rubinius/events{/privacy}',
      received_events_url:
        'https://api.github.com/users/rubinius/received_events',
      type: 'Organization',
      site_admin: false
    },
    private: false,
    html_url: 'https://github.com/rubinius/rubinius',
    description: 'The Rubinius Language Platform',
    fork: false,
    url: 'https://api.github.com/repos/rubinius/rubinius',
    forks_url: 'https://api.github.com/repos/rubinius/rubinius/forks',
    keys_url: 'https://api.github.com/repos/rubinius/rubinius/keys{/key_id}',
    collaborators_url:
      'https://api.github.com/repos/rubinius/rubinius/collaborators{/collaborator}',
    teams_url: 'https://api.github.com/repos/rubinius/rubinius/teams',
    hooks_url: 'https://api.github.com/repos/rubinius/rubinius/hooks',
    issue_events_url:
      'https://api.github.com/repos/rubinius/rubinius/issues/events{/number}',
    events_url: 'https://api.github.com/repos/rubinius/rubinius/events',
    assignees_url:
      'https://api.github.com/repos/rubinius/rubinius/assignees{/user}',
    branches_url:
      'https://api.github.com/repos/rubinius/rubinius/branches{/branch}',
    tags_url: 'https://api.github.com/repos/rubinius/rubinius/tags',
    blobs_url: 'https://api.github.com/repos/rubinius/rubinius/git/blobs{/sha}',
    git_tags_url:
      'https://api.github.com/repos/rubinius/rubinius/git/tags{/sha}',
    git_refs_url:
      'https://api.github.com/repos/rubinius/rubinius/git/refs{/sha}',
    trees_url: 'https://api.github.com/repos/rubinius/rubinius/git/trees{/sha}',
    statuses_url:
      'https://api.github.com/repos/rubinius/rubinius/statuses/{sha}',
    languages_url: 'https://api.github.com/repos/rubinius/rubinius/languages',
    stargazers_url: 'https://api.github.com/repos/rubinius/rubinius/stargazers',
    contributors_url:
      'https://api.github.com/repos/rubinius/rubinius/contributors',
    subscribers_url:
      'https://api.github.com/repos/rubinius/rubinius/subscribers',
    subscription_url:
      'https://api.github.com/repos/rubinius/rubinius/subscription',
    commits_url: 'https://api.github.com/repos/rubinius/rubinius/commits{/sha}',
    git_commits_url:
      'https://api.github.com/repos/rubinius/rubinius/git/commits{/sha}',
    comments_url:
      'https://api.github.com/repos/rubinius/rubinius/comments{/number}',
    issue_comment_url:
      'https://api.github.com/repos/rubinius/rubinius/issues/comments{/number}',
    contents_url:
      'https://api.github.com/repos/rubinius/rubinius/contents/{+path}',
    compare_url:
      'https://api.github.com/repos/rubinius/rubinius/compare/{base}...{head}',
    merges_url: 'https://api.github.com/repos/rubinius/rubinius/merges',
    archive_url:
      'https://api.github.com/repos/rubinius/rubinius/{archive_format}{/ref}',
    downloads_url: 'https://api.github.com/repos/rubinius/rubinius/downloads',
    issues_url:
      'https://api.github.com/repos/rubinius/rubinius/issues{/number}',
    pulls_url: 'https://api.github.com/repos/rubinius/rubinius/pulls{/number}',
    milestones_url:
      'https://api.github.com/repos/rubinius/rubinius/milestones{/number}',
    notifications_url:
      'https://api.github.com/repos/rubinius/rubinius/notifications{?since,all,participating}',
    labels_url: 'https://api.github.com/repos/rubinius/rubinius/labels{/name}',
    releases_url:
      'https://api.github.com/repos/rubinius/rubinius/releases{/id}',
    deployments_url:
      'https://api.github.com/repos/rubinius/rubinius/deployments'
  },
  {
    id: 31,
    name: 'jspec',
    full_name: 'wycats/jspec',
    owner: {
      login: 'wycats',
      id: 4,
      avatar_url: 'https://avatars0.githubusercontent.com/u/4?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/wycats',
      html_url: 'https://github.com/wycats',
      followers_url: 'https://api.github.com/users/wycats/followers',
      following_url:
        'https://api.github.com/users/wycats/following{/other_user}',
      gists_url: 'https://api.github.com/users/wycats/gists{/gist_id}',
      starred_url: 'https://api.github.com/users/wycats/starred{/owner}{/repo}',
      subscriptions_url: 'https://api.github.com/users/wycats/subscriptions',
      organizations_url: 'https://api.github.com/users/wycats/orgs',
      repos_url: 'https://api.github.com/users/wycats/repos',
      events_url: 'https://api.github.com/users/wycats/events{/privacy}',
      received_events_url:
        'https://api.github.com/users/wycats/received_events',
      type: 'User',
      site_admin: false
    },
    private: false,
    html_url: 'https://github.com/wycats/jspec',
    description: 'A JavaScript BDD Testing Library',
    fork: false,
    url: 'https://api.github.com/repos/wycats/jspec',
    forks_url: 'https://api.github.com/repos/wycats/jspec/forks',
    keys_url: 'https://api.github.com/repos/wycats/jspec/keys{/key_id}',
    collaborators_url:
      'https://api.github.com/repos/wycats/jspec/collaborators{/collaborator}',
    teams_url: 'https://api.github.com/repos/wycats/jspec/teams',
    hooks_url: 'https://api.github.com/repos/wycats/jspec/hooks',
    issue_events_url:
      'https://api.github.com/repos/wycats/jspec/issues/events{/number}',
    events_url: 'https://api.github.com/repos/wycats/jspec/events',
    assignees_url: 'https://api.github.com/repos/wycats/jspec/assignees{/user}',
    branches_url: 'https://api.github.com/repos/wycats/jspec/branches{/branch}',
    tags_url: 'https://api.github.com/repos/wycats/jspec/tags',
    blobs_url: 'https://api.github.com/repos/wycats/jspec/git/blobs{/sha}',
    git_tags_url: 'https://api.github.com/repos/wycats/jspec/git/tags{/sha}',
    git_refs_url: 'https://api.github.com/repos/wycats/jspec/git/refs{/sha}',
    trees_url: 'https://api.github.com/repos/wycats/jspec/git/trees{/sha}',
    statuses_url: 'https://api.github.com/repos/wycats/jspec/statuses/{sha}',
    languages_url: 'https://api.github.com/repos/wycats/jspec/languages',
    stargazers_url: 'https://api.github.com/repos/wycats/jspec/stargazers',
    contributors_url: 'https://api.github.com/repos/wycats/jspec/contributors',
    subscribers_url: 'https://api.github.com/repos/wycats/jspec/subscribers',
    subscription_url: 'https://api.github.com/repos/wycats/jspec/subscription',
    commits_url: 'https://api.github.com/repos/wycats/jspec/commits{/sha}',
    git_commits_url:
      'https://api.github.com/repos/wycats/jspec/git/commits{/sha}',
    comments_url: 'https://api.github.com/repos/wycats/jspec/comments{/number}',
    issue_comment_url:
      'https://api.github.com/repos/wycats/jspec/issues/comments{/number}',
    contents_url: 'https://api.github.com/repos/wycats/jspec/contents/{+path}',
    compare_url:
      'https://api.github.com/repos/wycats/jspec/compare/{base}...{head}',
    merges_url: 'https://api.github.com/repos/wycats/jspec/merges',
    archive_url:
      'https://api.github.com/repos/wycats/jspec/{archive_format}{/ref}',
    downloads_url: 'https://api.github.com/repos/wycats/jspec/downloads',
    issues_url: 'https://api.github.com/repos/wycats/jspec/issues{/number}',
    pulls_url: 'https://api.github.com/repos/wycats/jspec/pulls{/number}',
    milestones_url:
      'https://api.github.com/repos/wycats/jspec/milestones{/number}',
    notifications_url:
      'https://api.github.com/repos/wycats/jspec/notifications{?since,all,participating}',
    labels_url: 'https://api.github.com/repos/wycats/jspec/labels{/name}',
    releases_url: 'https://api.github.com/repos/wycats/jspec/releases{/id}',
    deployments_url: 'https://api.github.com/repos/wycats/jspec/deployments'
  },
  {
    id: 35,
    name: 'exception_logger',
    full_name: 'defunkt/exception_logger',
    owner: {
      login: 'defunkt',
      id: 2,
      avatar_url: 'https://avatars0.githubusercontent.com/u/2?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/defunkt',
      html_url: 'https://github.com/defunkt',
      followers_url: 'https://api.github.com/users/defunkt/followers',
      following_url:
        'https://api.github.com/users/defunkt/following{/other_user}',
      gists_url: 'https://api.github.com/users/defunkt/gists{/gist_id}',
      starred_url:
        'https://api.github.com/users/defunkt/starred{/owner}{/repo}',
      subscriptions_url: 'https://api.github.com/users/defunkt/subscriptions',
      organizations_url: 'https://api.github.com/users/defunkt/orgs',
      repos_url: 'https://api.github.com/users/defunkt/repos',
      events_url: 'https://api.github.com/users/defunkt/events{/privacy}',
      received_events_url:
        'https://api.github.com/users/defunkt/received_events',
      type: 'User',
      site_admin: true
    },
    private: false,
    html_url: 'https://github.com/defunkt/exception_logger',
    description: 'Unmaintained. Sorry.',
    fork: false,
    url: 'https://api.github.com/repos/defunkt/exception_logger',
    forks_url: 'https://api.github.com/repos/defunkt/exception_logger/forks',
    keys_url:
      'https://api.github.com/repos/defunkt/exception_logger/keys{/key_id}',
    collaborators_url:
      'https://api.github.com/repos/defunkt/exception_logger/collaborators{/collaborator}',
    teams_url: 'https://api.github.com/repos/defunkt/exception_logger/teams',
    hooks_url: 'https://api.github.com/repos/defunkt/exception_logger/hooks',
    issue_events_url:
      'https://api.github.com/repos/defunkt/exception_logger/issues/events{/number}',
    events_url: 'https://api.github.com/repos/defunkt/exception_logger/events',
    assignees_url:
      'https://api.github.com/repos/defunkt/exception_logger/assignees{/user}',
    branches_url:
      'https://api.github.com/repos/defunkt/exception_logger/branches{/branch}',
    tags_url: 'https://api.github.com/repos/defunkt/exception_logger/tags',
    blobs_url:
      'https://api.github.com/repos/defunkt/exception_logger/git/blobs{/sha}',
    git_tags_url:
      'https://api.github.com/repos/defunkt/exception_logger/git/tags{/sha}',
    git_refs_url:
      'https://api.github.com/repos/defunkt/exception_logger/git/refs{/sha}',
    trees_url:
      'https://api.github.com/repos/defunkt/exception_logger/git/trees{/sha}',
    statuses_url:
      'https://api.github.com/repos/defunkt/exception_logger/statuses/{sha}',
    languages_url:
      'https://api.github.com/repos/defunkt/exception_logger/languages',
    stargazers_url:
      'https://api.github.com/repos/defunkt/exception_logger/stargazers',
    contributors_url:
      'https://api.github.com/repos/defunkt/exception_logger/contributors',
    subscribers_url:
      'https://api.github.com/repos/defunkt/exception_logger/subscribers',
    subscription_url:
      'https://api.github.com/repos/defunkt/exception_logger/subscription',
    commits_url:
      'https://api.github.com/repos/defunkt/exception_logger/commits{/sha}',
    git_commits_url:
      'https://api.github.com/repos/defunkt/exception_logger/git/commits{/sha}',
    comments_url:
      'https://api.github.com/repos/defunkt/exception_logger/comments{/number}',
    issue_comment_url:
      'https://api.github.com/repos/defunkt/exception_logger/issues/comments{/number}',
    contents_url:
      'https://api.github.com/repos/defunkt/exception_logger/contents/{+path}',
    compare_url:
      'https://api.github.com/repos/defunkt/exception_logger/compare/{base}...{head}',
    merges_url: 'https://api.github.com/repos/defunkt/exception_logger/merges',
    archive_url:
      'https://api.github.com/repos/defunkt/exception_logger/{archive_format}{/ref}',
    downloads_url:
      'https://api.github.com/repos/defunkt/exception_logger/downloads',
    issues_url:
      'https://api.github.com/repos/defunkt/exception_logger/issues{/number}',
    pulls_url:
      'https://api.github.com/repos/defunkt/exception_logger/pulls{/number}',
    milestones_url:
      'https://api.github.com/repos/defunkt/exception_logger/milestones{/number}',
    notifications_url:
      'https://api.github.com/repos/defunkt/exception_logger/notifications{?since,all,participating}',
    labels_url:
      'https://api.github.com/repos/defunkt/exception_logger/labels{/name}',
    releases_url:
      'https://api.github.com/repos/defunkt/exception_logger/releases{/id}',
    deployments_url:
      'https://api.github.com/repos/defunkt/exception_logger/deployments'
  },
  {
    id: 36,
    name: 'ambition',
    full_name: 'defunkt/ambition',
    owner: {
      login: 'defunkt',
      id: 2,
      avatar_url: 'https://avatars0.githubusercontent.com/u/2?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/defunkt',
      html_url: 'https://github.com/defunkt',
      followers_url: 'https://api.github.com/users/defunkt/followers',
      following_url:
        'https://api.github.com/users/defunkt/following{/other_user}',
      gists_url: 'https://api.github.com/users/defunkt/gists{/gist_id}',
      starred_url:
        'https://api.github.com/users/defunkt/starred{/owner}{/repo}',
      subscriptions_url: 'https://api.github.com/users/defunkt/subscriptions',
      organizations_url: 'https://api.github.com/users/defunkt/orgs',
      repos_url: 'https://api.github.com/users/defunkt/repos',
      events_url: 'https://api.github.com/users/defunkt/events{/privacy}',
      received_events_url:
        'https://api.github.com/users/defunkt/received_events',
      type: 'User',
      site_admin: true
    },
    private: false,
    html_url: 'https://github.com/defunkt/ambition',
    description: 'include Enumerable — Unmaintained',
    fork: false,
    url: 'https://api.github.com/repos/defunkt/ambition',
    forks_url: 'https://api.github.com/repos/defunkt/ambition/forks',
    keys_url: 'https://api.github.com/repos/defunkt/ambition/keys{/key_id}',
    collaborators_url:
      'https://api.github.com/repos/defunkt/ambition/collaborators{/collaborator}',
    teams_url: 'https://api.github.com/repos/defunkt/ambition/teams',
    hooks_url: 'https://api.github.com/repos/defunkt/ambition/hooks',
    issue_events_url:
      'https://api.github.com/repos/defunkt/ambition/issues/events{/number}',
    events_url: 'https://api.github.com/repos/defunkt/ambition/events',
    assignees_url:
      'https://api.github.com/repos/defunkt/ambition/assignees{/user}',
    branches_url:
      'https://api.github.com/repos/defunkt/ambition/branches{/branch}',
    tags_url: 'https://api.github.com/repos/defunkt/ambition/tags',
    blobs_url: 'https://api.github.com/repos/defunkt/ambition/git/blobs{/sha}',
    git_tags_url:
      'https://api.github.com/repos/defunkt/ambition/git/tags{/sha}',
    git_refs_url:
      'https://api.github.com/repos/defunkt/ambition/git/refs{/sha}',
    trees_url: 'https://api.github.com/repos/defunkt/ambition/git/trees{/sha}',
    statuses_url:
      'https://api.github.com/repos/defunkt/ambition/statuses/{sha}',
    languages_url: 'https://api.github.com/repos/defunkt/ambition/languages',
    stargazers_url: 'https://api.github.com/repos/defunkt/ambition/stargazers',
    contributors_url:
      'https://api.github.com/repos/defunkt/ambition/contributors',
    subscribers_url:
      'https://api.github.com/repos/defunkt/ambition/subscribers',
    subscription_url:
      'https://api.github.com/repos/defunkt/ambition/subscription',
    commits_url: 'https://api.github.com/repos/defunkt/ambition/commits{/sha}',
    git_commits_url:
      'https://api.github.com/repos/defunkt/ambition/git/commits{/sha}',
    comments_url:
      'https://api.github.com/repos/defunkt/ambition/comments{/number}',
    issue_comment_url:
      'https://api.github.com/repos/defunkt/ambition/issues/comments{/number}',
    contents_url:
      'https://api.github.com/repos/defunkt/ambition/contents/{+path}',
    compare_url:
      'https://api.github.com/repos/defunkt/ambition/compare/{base}...{head}',
    merges_url: 'https://api.github.com/repos/defunkt/ambition/merges',
    archive_url:
      'https://api.github.com/repos/defunkt/ambition/{archive_format}{/ref}',
    downloads_url: 'https://api.github.com/repos/defunkt/ambition/downloads',
    issues_url: 'https://api.github.com/repos/defunkt/ambition/issues{/number}',
    pulls_url: 'https://api.github.com/repos/defunkt/ambition/pulls{/number}',
    milestones_url:
      'https://api.github.com/repos/defunkt/ambition/milestones{/number}',
    notifications_url:
      'https://api.github.com/repos/defunkt/ambition/notifications{?since,all,participating}',
    labels_url: 'https://api.github.com/repos/defunkt/ambition/labels{/name}',
    releases_url: 'https://api.github.com/repos/defunkt/ambition/releases{/id}',
    deployments_url: 'https://api.github.com/repos/defunkt/ambition/deployments'
  }
];
