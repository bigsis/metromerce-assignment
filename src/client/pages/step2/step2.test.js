import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import fetchMock from 'fetch-mock';
import Step2 from './step2';
import { mockRepo } from './mock';

fetchMock.get('*', mockRepo);
describe('Step2 page', () => {
  it('should be defined', () => {
    expect(Step2).toBeDefined();
  });

  it('should match snapshot when format json', () => {
    const wrapper = mount(<Step2 />, { attachTo: document.body });

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
