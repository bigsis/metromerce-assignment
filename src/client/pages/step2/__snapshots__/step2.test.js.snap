// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step2 page should match snapshot when format json 1`] = `
<Step2>
  <div
    className="step2-content"
  >
    <Table
      responsiveTag="div"
      tag="table"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              ID
            </th>
            <th>
              Avatar
            </th>
            <th>
              User Name
            </th>
            <th>
              Repo Name
            </th>
            <th>
              Repo description
            </th>
            <th>
              Repo Url
            </th>
          </tr>
        </thead>
        <tbody>
          <h1>
            Loading.. please wait!
          </h1>
        </tbody>
      </table>
    </Table>
    <Pagination
      tag="ul"
    >
      <ul
        className="pagination"
      >
        <PaginationItem
          active={true}
          key="1"
          tag="li"
        >
          <li
            className="page-item active"
          >
            <PaginationLink
              id="pagination-link-1"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-1"
                onClick={[Function]}
              >
                1
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
        <PaginationItem
          active={false}
          key="2"
          tag="li"
        >
          <li
            className="page-item"
          >
            <PaginationLink
              id="pagination-link-2"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-2"
                onClick={[Function]}
              >
                2
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
        <PaginationItem
          active={false}
          key="3"
          tag="li"
        >
          <li
            className="page-item"
          >
            <PaginationLink
              id="pagination-link-3"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-3"
                onClick={[Function]}
              >
                3
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
        <PaginationItem
          active={false}
          key="4"
          tag="li"
        >
          <li
            className="page-item"
          >
            <PaginationLink
              id="pagination-link-4"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-4"
                onClick={[Function]}
              >
                4
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
        <PaginationItem
          active={false}
          key="5"
          tag="li"
        >
          <li
            className="page-item"
          >
            <PaginationLink
              id="pagination-link-5"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-5"
                onClick={[Function]}
              >
                5
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
        <PaginationItem
          active={false}
          key="6"
          tag="li"
        >
          <li
            className="page-item"
          >
            <PaginationLink
              id="pagination-link-6"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-6"
                onClick={[Function]}
              >
                6
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
        <PaginationItem
          active={false}
          key="7"
          tag="li"
        >
          <li
            className="page-item"
          >
            <PaginationLink
              id="pagination-link-7"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-7"
                onClick={[Function]}
              >
                7
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
        <PaginationItem
          active={false}
          key="8"
          tag="li"
        >
          <li
            className="page-item"
          >
            <PaginationLink
              id="pagination-link-8"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-8"
                onClick={[Function]}
              >
                8
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
        <PaginationItem
          active={false}
          key="9"
          tag="li"
        >
          <li
            className="page-item"
          >
            <PaginationLink
              id="pagination-link-9"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-9"
                onClick={[Function]}
              >
                9
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
        <PaginationItem
          active={false}
          key="10"
          tag="li"
        >
          <li
            className="page-item"
          >
            <PaginationLink
              id="pagination-link-10"
              onClick={[Function]}
              tag="a"
            >
              <a
                className="page-link"
                id="pagination-link-10"
                onClick={[Function]}
              >
                10
              </a>
            </PaginationLink>
          </li>
        </PaginationItem>
      </ul>
    </Pagination>
  </div>
</Step2>
`;
